light-rpc/src/main.rs

```diff
@@ -1,4 +1,503 @@
+                    commitment: CommitmentLevel::Finalized,
+                }),
+                encoding: None,
+                transaction_details: Some(
+                    solana_transaction_status::TransactionDetails::Signatures,
+                ),
+                show_rewards: None,
+                max_supported_transaction_version: None,
+            }),
+        )
+        .unwrap();
+
+        let mut joinables = Vec::new();
+        let signature_status = Arc::new(RwLock::new(HashMap::new()));
+
+        let confirmation_slot = rpc_client
+            .get_slot_with_commitment(CommitmentConfig {
+                commitment: CommitmentLevel::Confirmed,
+            })
+            .unwrap();
+        let finalized_slot = rpc_client
+            .get_slot_with_commitment(CommitmentConfig {
+                commitment: CommitmentLevel::Finalized,
+            })
+            .unwrap();
+
+        let confirmation_slot = Arc::new(AtomicU64::new(confirmation_slot));
+        let finalized_slot = Arc::new(AtomicU64::new(finalized_slot));
+
+        // start thread which will listen to confirmed blocks
+        {
+            let signature_status = signature_status.clone();
+            let confirmation_slot = confirmation_slot.clone();
+            let finalized_slot = finalized_slot.clone();
+            let join_handle = Builder::new()
+                .name("thread working on confirmation block".to_string())
+                .spawn(move || {
+                    Self::process_block(
+                        receiver_confirmed,
+                        signature_status,
+                        CommitmentLevel::Confirmed,
+                        confirmation_slot,
+                    );
+                })
+                .unwrap();
+            joinables.push(join_handle);
+        }
+        // start thread which will listen to finalized blocks
+        {
+            let signature_status = signature_status.clone();
+            let confirmation_slot = confirmation_slot.clone();
+            let finalized_slot = finalized_slot.clone();
+            let join_handle = Builder::new()
+                .name("thread working on finalized block".to_string())
+                .spawn(move || {
+                    Self::process_block(
+                        receiver_finalized,
+                        signature_status,
+                        CommitmentLevel::Finalized,
+                        finalized_slot,
+                    );
+                })
+                .unwrap();
+            joinables.push(join_handle);
+        }
+
+        LightRpcRequestProcessor {
+            rpc_client,
+            tpu_client,
+            last_valid_block_height: 0,
+            ws_url: websocket_url.clone(),
+            current_slot,
+            signature_status,
+            connection_cache: connection_cache,
+            subscribed_clients: Arc::new(vec![client_confirmed, client_finalized]),
```

Comment:
suggest to avoid the code duplication for xyz_confirmed + xyz_finalized that happens before this point; instead the setup code for each configuration could go into one place.

Reply:
I did not understand you suggestion.